### PR TITLE
VB init-only: Conversions in init-only phase

### PR DESCRIPTION
**Pull request by AlekseyTs**
This is a VB "counterpart" of https://github.com/dotnet/roslyn/pull/50424.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@jcouv, @dotnet/roslyn-compiler Please review.

**Review by 333fred — APPROVED** [review]

**Review by jcouv — DISMISSED** [review]
LGTM Thanks (iteration 2)

**Comment by jaredpar** [comment]
@AlekseyTs can we re-target this to `release/dev16.9`?

**Comment by AlekseyTs** [comment]
> can we re-target this to release/dev16.9?

Done

**Comment by AlekseyTs** [comment]
@333fred, @jcouv Please review Additional changes made to reflect LDM decision.

**Review by 333fred — APPROVED** [review]

**Comment by AlekseyTs** [comment]
@jcouv Could you please take another look?

**Comment by AlekseyTs** [comment]
@jcouv Could you please take another look?

**Comment by AlekseyTs** [comment]
@jaredpar Do we need to get any special sign-off to merge into release/dev16.9?

File: src/Compilers/VisualBasic/Portable/Symbols/PropertySymbol.vb [hunk 1]
```diff
@@ -175,28 +175,34 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
...
-                    Case BoundKind.MeReference, BoundKind.MyBaseReference, BoundKind.MyClassReference
-                        Return True
-                    Case Else
-                        Return False
-                End Select
+                Do
+                    Select Case receiverOpt.Kind
+                        Case BoundKind.MeReference, BoundKind.MyBaseReference, BoundKind.MyClassReference
+                            Return True
+                        Case BoundKind.Parenthesized
```
Comment (jcouv) [inline src/Compilers/VisualBasic/Portable/Symbols/PropertySymbol.vb:200]: Looks like this fixes the handling for parenthesized receiver in `With`. Do we have a test for that case?

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 4) with a test question (nit)

File: src/Compilers/VisualBasic/Portable/Symbols/PropertySymbol.vb (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/VisualBasic/Portable/Symbols/PropertySymbol.vb:200]: >Looks like this fixes the handling for parenthesized receiver in With. Do we have a test for that case?

EvaluationInitOnlySetter_10

---
In reply to: [566277477](https://github.com/dotnet/roslyn/pull/50770#discussion_r566277477) [](ancestors = 566277477)

**Comment by AlekseyTs** [comment]
@jaredpar Good to merge?

**Pull request merged**